[mdoc] CI job breaks for .NET assemblies

When running `mdoc` on existing .NET platform assemblies, the tool errors out with:

```
2017-12-14T19:06:51.4374595Z Framework configuration file written to D:\a\1\s\source_repo\dotnet\frameworks.xml
2017-12-14T19:06:51.6418588Z Opening frameworks file 'D:\a\1\s\source_repo\dotnet\frameworks.xml'
2017-12-14T19:06:51.7783602Z Frameworks Configuration contains 1824 assemblies
2017-12-14T19:06:57.5354394Z ##[error]mdoc: System.NotSupportedException: System.Windows.RoutedEvent System.Windows.DataObject::Copying
   at Mono.Cecil.Rocks.DocCommentId.GetDocCommentId(IMemberDefinition member)
   at Mono.Documentation.Updater.Frameworks.FrameworkTypeEntry.ProcessMember(MemberReference member)
   at Mono.Documentation.MDocUpdater.Run(IEnumerable`1 args)
   at Mono.Documentation.MDoc.Run(String[] args)
   at Mono.Documentation.MDoc.Main(String[] args)
See `mdoc help' for more information or use --debug to diagnose.
```

--- a/mdoc/Consts.cs
+++ b/mdoc/Consts.cs
@@ -3,7 +3,7 @@ namespace Mono.Documentation
 {
 	public static class Consts
 	{
-		public static string MonoVersion = "5.3.5.1";
+		public static string MonoVersion = "5.3.5.2";
 		public const string DocId = "DocId";
 		public const string VbNet = "VB.NET";
 		public const string DocIdLowCase = "docid";

--- a/mdoc/Mono.Documentation/MDocUpdater.cs
+++ b/mdoc/Mono.Documentation/MDocUpdater.cs
@@ -2156,11 +2156,15 @@ namespace Mono.Documentation
                 var explicitTypeName = DocUtils.GetExplicitTypeName(mi);
 
                 // find one member wich is pocessed by the explicitly mentioned type
-                var explicitlyImplemented = implementedMembers.First(i => i.DeclaringType.GetElementType().FullName == explicitTypeName);
-                implementedMembers = new List<MemberReference>
+                var explicitlyImplemented = implementedMembers.FirstOrDefault(i => i.DeclaringType.GetElementType().FullName == explicitTypeName);
+
+                if (explicitlyImplemented != null)
                 {
-                    explicitlyImplemented
-                };
+                    implementedMembers = new List<MemberReference>
+                    {
+                        explicitlyImplemented
+                    };
+                }
             }
 
             if (!implementedMembers.Any())

--- a/mdoc/Mono.Documentation/Updater/Frameworks/FrameworkTypeEntry.cs
+++ b/mdoc/Mono.Documentation/Updater/Frameworks/FrameworkTypeEntry.cs
@@ -12,6 +12,7 @@ namespace Mono.Documentation.Updater.Frameworks
         Dictionary<string, bool> sigMap = new Dictionary<string, bool> ();
 
 		ILFullMemberFormatter formatter = new ILFullMemberFormatter ();
+        DocIdFormatter docidFormatter = new DocIdFormatter ();
 
 		FrameworkEntry fx;
 
@@ -37,7 +38,7 @@ namespace Mono.Documentation.Updater.Frameworks
 		{
 			var resolvedMember = member.Resolve ();
 			if (resolvedMember != null) {
-				var docid = DocCommentId.GetDocCommentId (resolvedMember);
+                var docid = docidFormatter.GetDeclaration (member);
 				members.Add (docid);
 			}
 			else 

--- a/mdoc/Mono.Documentation/Updater/Frameworks/MDocResolver.cs
+++ b/mdoc/Mono.Documentation/Updater/Frameworks/MDocResolver.cs
@@ -101,6 +101,7 @@ namespace Mono.Documentation.Updater.Frameworks
                 .Select (a => new
                 {
                     Assembly = a,
+                    SuppliedDependency = namedPaths.Any (np => np == a.MainModule.FileName),
                     VersionSort = aggregateVersion (a.Name.Version),
                     VersionDiff = aggregateVersion (a.Name.Version) - aggregateVersion (name.Version),
                     MajorMatches = a.Name.Version.Major == name.Version.Major
@@ -109,9 +110,13 @@ namespace Mono.Documentation.Updater.Frameworks
 
             // If the assembly has all zeroes, just grab the latest assembly
             if (IsZero(name.Version)) {
-                var highestMatch = applicableVersions
-                    .OrderByDescending (v => v.VersionSort)
-                    .FirstOrDefault ();
+                var possibleSet = applicableVersions;
+                if (applicableVersions.Any (s => s.SuppliedDependency))
+                    possibleSet = applicableVersions.Where (av => av.SuppliedDependency).ToArray ();
+                
+                var sorted = possibleSet.OrderByDescending (v => v.VersionSort).ToArray ();
+
+                var highestMatch = sorted.FirstOrDefault ();
                 if (highestMatch != null)
                     return highestMatch.Assembly;
             }

--- a/mdoc/mdoc.nuspec
+++ b/mdoc/mdoc.nuspec
@@ -2,7 +2,7 @@
 <package >
   <metadata>
     <id>mdoc</id>
-    <version>5.3.5.1</version>
+    <version>5.3.5.2</version>
     <title>mdoc</title>
     <authors>Microsoft</authors>
     <owners>Microsoft</owners>

